iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/CertificateBasedAuthenticationCheck.java

```diff
@@ -19,35 +19,69 @@
  */
 package org.sonar.iac.arm.checks;
 
+import java.util.LinkedList;
+import java.util.List;
+import org.sonar.iac.arm.tree.ArmTreeUtils;
 import org.sonar.iac.arm.tree.api.ArmTree;
 import org.sonar.iac.arm.tree.api.BooleanLiteral;
 import org.sonar.iac.arm.tree.api.ResourceDeclaration;
+import org.sonar.iac.arm.tree.api.StringLiteral;
 import org.sonar.iac.common.api.checks.CheckContext;
 import org.sonar.iac.common.api.tree.Tree;
 import org.sonar.iac.common.checks.PropertyUtils;
 
 public class CertificateBasedAuthenticationCheck extends AbstractArmResourceCheck {
 
-  private static final String MISSING_CERTIFICATE = "Omitting \"%s\" disables certificate-based authentication. Make sure it is safe here.";
-  private static final String DISABLED_CERTIFICATE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String MISSING_CERTIFICATE_MESSAGE = "Omitting \"%s\" disables certificate-based authentication. Make sure it is safe here.";
+  private static final String DISABLED_CERTIFICATE_MESSAGE = "Make sure that disabling certificate-based authentication is safe here.";
+  private static final String ALLOWING_NO_CERTIFICATE_MESSAGE = "Connections without client certificates will be permitted. Make sure it is safe here.";
+
+  private static final List<String> PATH_CONTAINER_APPS = ArmTreeUtils.computePath("configuration/ingress");
 
   @Override
   protected void registerResourceConsumer() {
-    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkNegotiateClientCertificate);
+    register("Microsoft.ApiManagement/service/gateways/hostnameConfigurations", CertificateBasedAuthenticationCheck::checkHostnameConfigurations);
+    register("Microsoft.App/containerApps", CertificateBasedAuthenticationCheck::checkContainerApps);
   }
 
-  private static void checkNegotiateClientCertificate(CheckContext ctx, ResourceDeclaration resource) {
-    PropertyUtils.value(resource, "negotiateClientCertificate")
+  private static void checkHostnameConfigurations(CheckContext ctx, ResourceDeclaration resource) {
+    PropertyUtils.get(resource, "negotiateClientCertificate")
       .ifPresentOrElse(
-        value -> {
-          if (isBooleanFalse(value)) {
-            ctx.reportIssue(value, DISABLED_CERTIFICATE);
+        property -> {
+          if (isBooleanFalse(property.value())) {
+            ctx.reportIssue(property, DISABLED_CERTIFICATE_MESSAGE);
           }
         },
-        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE, "negotiateClientCertificate")));
+        () -> ctx.reportIssue(resource.type(), String.format(MISSING_CERTIFICATE_MESSAGE, "negotiateClientCertificate")));
+  }
```

Comment:
When looking at this, I wonder if we actually use the DSL of the Terraform checks. That way we would write this logic like this:
```
resource.attribute("negotiateClientCertificate")
 .reportIf(isFalse(), DISABLED_CERTIFICATE_MESSAGE)
 .reportIfAbsent(MISSING_CERTIFICATE_MESSAGE)
```
Only a comment, no action is needed here. But maybe we can discuss it.

Reply:
I'm currently not satisfied with the readability of this check.
The syntax you are proposing looks really well and easy to read, would love to be able to use it!

Still, I am working on the same main tickets and have implemented about 4 more other resources check.
I encountered several cases where we need to report an issue only when multiple conditions on different properties are true, like:
- `resource.properties.activities.*.type` is `WebActivity` or `WebHook`
- `resource.properties.activities.*.typeProperties.authenticationType` is `Basic` or `ServicePrincipal`

So we would need to be able to adapt it over time to support such use cases, not sure what are the current possibilities of DSL Terraform checks currently in place. But definitely it would bring a lot of improvement in writing checks.